### ISSUE TITLE
Comment back in Ca, Mg, and K cations in aerosol thermodynamics

### ISSUE THREAD

**Issue by msulprizio** [geoschem/geos-chem#2391]
### Your name

Melissa Sulprizio

### Your affiliation

Harvard

### Provide a clear and concise overview of the new feature requested.

These species are commented out in ISORROPIA and HETP and should be uncommented.

Tagging @beckyalexander for more details and a better estimate of delivery timeframe.

### Will you be implementing this feature yourself?

No

**Comment by lizziel** [comment]
@msulprizio you beat me to creating this. I discussed this issue with @beckyalexander and I volunteered to make the update. I can put it into 14.5.0.

**Comment by yantosca** [comment]
@msulprizio @lizziel @beckyalexander: The benchmark plots are available at https://gc-dashboard.org/difference?primary_key=diff-gcc-4x5-1Mon-14.4.2-gcc-4x5-1Mon-14.5.0-alpha.2

**Comment by msulprizio** [comment]
This has been addressed by #2398

**Comment by yc-chan** [comment]
@beckyalexander and I found a potential mass-balance issue in this implementation.

To account for the missing Ca, Mg, and K, the Na-to-seasalt mass ratio was artificially increased from 0.3061 (i.e. seawater average) to 0.397 in previous versions. 

Since we are re-introducing Ca, Mg, and K to the model now, we should adjust the Na-to-SS mass ratio from 0.397 back to 0.3061. Otherwise, we would get excess amount of cations in the model.

**Comment by lizziel** [comment]
Thanks for reporting this. Can you confirm this is the code that is affected? https://github.com/geoschem/geos-chem/blob/bef56c605e018eecbd91646a51ce82c7cd77f56a/GeosCore/aerosol_thermodynamics_mod.F90#L639-L646

Also please confirm the fix is as follows:
1. Delete the currently uncommented code which uses 0.397
2. Uncomment the currently commented out code that uses 0.3061

Note that the equations are different in other ways than the scale factor. Please let us know what the correct equation should be.

**Comment by lizziel** [comment]
@yc-chan, @beckyalexander - We plan to freeze 14.5.0 in the next few days and begin benchmarking. Please let us know exactly what the fix is as soon as possible to ensure it gets into this version. Otherwise it will need to wait for 14.6.0.

**Comment by beckyalexander** [comment]
I think you should change the value of 0.397 to 0.3061. There are other differences in the two equations (such as the use of the AlkR variable). All we want to change right now is the mass fraction of Na in sea salt. This equation is for fine-mode sea salt.

I also think line 658 needs to be changed, which deals with coarse-mode sea salt aerosol. The current value is 0.378. I don't know why these values are different for fine and coarse mode. They should both be 0.3061, which is the dry mass fraction of sea salt.

**Comment by msulprizio** [comment]
The additional issue brought forth by @yc-chan [above](https://github.com/geoschem/geos-chem/issues/2391#issuecomment-2356640509) has now been addressed by #2467 which has been merged into the `dev/14.5.0` branch.

**Comment by yc-chan** [comment]
@beckyalexander and I discovered another issue as we are testing the new code. 

In the past, GC only called ISORROPIA for doing fine-mode-aerosol calculations, so it assumed all the Ca, Mg, and K in aerosols are from fine-mode seasalt. However, DO_ATE is now also used for coarse-mode-aerosol calculations so we have to update the code for that. 

In short, we should replace this block: 


```
          ! Total Ca2+ (1.16% by weight of seasalt) [mole/m3]
          TCA      = Spc(id_SALA)%Conc(I,J,L) * 0.0116e+0_fp * 1.d3 / &
                                     ( 40.08e+0_fp  * VOL  )

          ! Total K+   (1.1% by weight of seasalt)  [mole/m3]
          TK       = Spc(id_SALA)%Conc(I,J,L) * 0.0110e+0_fp * 1.d3 / &
                                     ( 39.102e+0_fp * VOL  )

          ! Total Mg+  (3.69% by weight of seasalt) [mole/m3]
          TMG      = Spc(id_SALA)%Conc(I,J,L) * 0.0369e+0_fp * 1.d3 / &
                                     ( 24.312e+0_fp * VOL  )
```

with something like this:

```
IF (N == 1) THEN

          ! Total Ca2+ (1.16% by weight of fine-mode seasalt) [mole/m3]
          TCA      = Spc(id_SALA)%Conc(I,J,L) * 0.0116e+0_fp * 1.0e+3_fp * AlkR / &
                                     ( 40.08e+0_fp  * VOL  )

          ! Total K+   (1.1% by weight of fine-mode seasalt)  [mole/m3]
          TK       = Spc(id_SALA)%Conc(I,J,L) * 0.0110e+0_fp * 1.0e+3_fp * AlkR / &
                                     ( 39.102e+0_fp * VOL  )

          ! Total Mg+  (3.69% by weight of fine-mode seasalt) [mole/m3]
          TMG      = Spc(id_SALA)%Conc(I,J,L) * 0.0369e+0_fp * 1.0e+3_fp * AlkR / &
                                     ( 24.312e+0_fp * VOL  )

ELSE

          ! Total Ca2+ (1.16% by weight of coarse-mode seasalt) [mole/m3]
          TCA      = Spc(id_SALC)%Conc(I,J,L) * 0.0116e+0_fp * 1.0e+3_fp * AlkR / &
                                     ( 40.08e+0_fp  * VOL  )

          ! Total K+   (1.1% by weight of coarse-mode seasalt)  [mole/m3]
          TK       = Spc(id_SALC)%Conc(I,J,L) * 0.0110e+0_fp * 1.0e+3_fp * AlkR / &
                                     ( 39.102e+0_fp * VOL  )

          ! Total Mg+  (3.69% by weight of coarse-mode seasalt) [mole/m3]
          TMG      = Spc(id_SALC)%Conc(I,J,L) * 0.0369e+0_fp * 1.0e+3_fp * AlkR / &
                                     ( 24.312e+0_fp * VOL  )

ENDIF
```

Note that the alkalinity factor AlkR is now included in the formula for TCA, TK, and TMG to make sure they are treated the same way as TNA.

Without this fix, I expect that GC would underestimate cation abundance in coarse-mode aerosols over the marine boundary layer, especially after the previous correction to Na-to-seasalt mass ratio.

**Comment by msulprizio** [comment]
Thanks for reporting this @yc-chan @beckyalexander. We are about to begin benchmarking for version 14.5.0 this week. If you could please [submit a pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) with this fix we can try to get it into this version.

**Comment by yc-chan** [comment]
I just submitted a [PR](https://github.com/geoschem/geos-chem/pull/2489). 

I don't have a strong opinion on whether the fix should be merged in 14.5.0 or 14.6.0. Tag @beckyalexander to weigh in.

**Comment by msulprizio** [comment]
Thanks @yc-chan. This latest PR has now been merged into the 14.5.0 development branch. We will proceed with benchmarking that version in preparation for release.
